scripts/bert/embedding.py

```diff
@@ -175,10 +207,13 @@ def oov(self, batches, oov_way='avg'):
                                      formatter_class=argparse.RawTextHelpFormatter)
     parser.add_argument('--gpu', type=int, default=None,
                         help='id of the gpu to use. Set it to empty means to use cpu.')
+    parser.add_argument('--dtype', type=str, default='float32', help='data dtype')
```

Comment:
@Deseaus Would you also add new arguments to the tests? [scripts/tests/test_bert_embedding.py](https://github.com/dmlc/gluon-nlp/blob/master/scripts/tests/test_bert_embedding.py)

Thank you!

Reply:
Good idea. @szha added the a test for the params path already and I added one for the `verbose` argument. All that's left is the `dtype` parameter. For that we'd have to fine-tune a model with `run_pretraining.py` and `float16`, and have the resulting checkpoint loaded in the embeddings test. I'm not quite sure how to go about that though: where the checkpoint should be stored, etc. Could you help out with that?

Reply:
Thank you @Deseaus .

We may add the option below first.  Making sure the command line interface works, which is more fit to the scenario here.

```diff
-    process = subprocess.check_call(['python', './scripts/bert/embedding.py', '--gpu', '0',
+    process = subprocess.check_call(['python', './scripts/bert/embedding.py', '--gpu', '0', '--dtype', 'float32',
                                     '--model', 'bert_12_768_12', '--dataset_name', 'book_corpus_wiki_en_uncased',
                                     '--max_seq_length', '25', '--batch_size', '256',
                                     '--oov_way', 'avg', '--sentences', '"is this jacksonville ?"'])
```

A different dtype pre-trained can be tested later on. (Also, it might be more appropriate to test in [/tests/unittest/test_models.py](https://github.com/dmlc/gluon-nlp/blob/master/tests/unittest/test_models.py). 

Does it make sense to you?